packages/artifact/src/internal/download-http-client.ts

```diff
@@ -66,16 +71,16 @@ export class DownloadHttpClient {
     const resourceUrl = new URL(containerUrl)
     resourceUrl.searchParams.append('itemPath', artifactName)
 
-    // use the first client from the httpManager, `keep-alive` is not used so the connection will close immediately
     const client = this.downloadHttpManager.getClient(0)
-    const requestOptions = getDownloadRequestOptions('application/json')
-    const response = await client.get(resourceUrl.toString(), requestOptions)
+    const headers = getDownloadHeaders('application/json', true)
+    const response = await client.get(resourceUrl.toString(), headers)
     const body: string = await response.readBody()
 
     if (isSuccessStatusCode(response.message.statusCode) && body) {
       return JSON.parse(body)
     }
     displayHttpDiagnostics(response)
+    this.disposeAllConnections()
```

Comment:
Why do you dispose of all clients here?  Don't we need those clients later? Are you recreating clients somewhere?

If those clients are single use, why not just create a HTTP client here and dispose of it when you are done.  Why bother with `HttpManager`?

Reply:
Right below this line an error is thrown, so the download essentially stops and no more HTTP calls get made (the clients don't get used anywhere else if this gets hit). I'm actually running some tests right now where we don't dispose at all, disposal might (should?) happen automatically when node finished up.